packages/backend/src/graphql/resolvers/receiver.test.ts

```diff
@@ -0,0 +1,254 @@
+import { gql } from 'apollo-server-koa'
+import { v4 as uuid } from 'uuid'
+import { createTestApp, TestContainer } from '../../tests/app'
+import { IocContract } from '@adonisjs/fold'
+import { AppServices } from '../../app'
+import { initIocContainer } from '../..'
+import { Config } from '../../config/app'
+import { Amount, serializeAmount } from '../../open_payments/amount'
+import { mockIncomingPayment, mockPaymentPointer } from 'open-payments'
+import { CreateReceiverResponse } from '../generated/graphql'
+import { ReceiverService } from '../../open_payments/receiver/service'
+import { Receiver } from '../../open_payments/receiver/model'
+import { ReceiverError } from '../../open_payments/receiver/errors'
+
+describe('Receiver Resolver', (): void => {
+  let deps: IocContract<AppServices>
+  let appContainer: TestContainer
+  let receiverService: ReceiverService
+
+  beforeAll(async (): Promise<void> => {
+    deps = initIocContainer(Config)
+    appContainer = await createTestApp(deps)
+    receiverService = await deps.use('receiverService')
+  })
+
+  afterAll(async (): Promise<void> => {
+    appContainer.apolloClient.stop()
+    await appContainer.shutdown()
+  })
+
+  describe('Mutation.createReceiver', (): void => {
+    const amount: Amount = {
+      value: BigInt(123),
+      assetCode: 'USD',
+      assetScale: 2
+    }
+    const paymentPointer = mockPaymentPointer()
+
+    test.each`
+      incomingAmount | expiresAt                        | description                | externalRef
+      ${undefined}   | ${undefined}                     | ${undefined}               | ${undefined}
+      ${amount}      | ${new Date(Date.now() + 30_000)} | ${'Test incoming payment'} | ${'#123'}
+    `(
+      'creates receiver ($#)',
+      async ({
+        description,
+        externalRef,
+        expiresAt,
+        incomingAmount
+      }): Promise<void> => {
+        const receiver = Receiver.fromIncomingPayment(
+          mockIncomingPayment({
+            id: `${paymentPointer.id}/incoming-payments/${uuid()}`,
+            paymentPointer: paymentPointer.id,
+            description,
+            externalRef,
+            expiresAt: expiresAt?.toISOString(),
+            incomingAmount: incomingAmount
+              ? serializeAmount(incomingAmount)
+              : undefined
+          })
+        ) as Receiver
```

Comment:
Would it be worth adding a `createReceiver` (here or in `tests/receiver`) that returns a guaranteed `Receiver`?

Reply:
Maybe it would be better to have `Receiver.fromIncomingPayment/fromConnection`to throw errors instead of returning undefined?

Reply:
:+1: that or return `Receiver | ReceiverError` :grimacing:

Reply:
Ended up going with just throwing, will likely be changing out how the Receiver class works anyway if we want to go with `type Receiver = IncomingPayment | Connection`